<a href="https://colab.research.google.com/github/chaitanysaraswat/Advanced-Analytics-Using-Statistics/blob/main/lecture10.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [8]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings("ignore")
pd.set_option('display.max_columns', None)
from numpy import random
from collections import Counter
from numpy.linalg import inv
from numpy.linalg import eig
import matplotlib
from matplotlib import pyplot as plt
import seaborn as sns
import pylab
from pylab import legend
from pylab import plot, show, title, xlabel, ylabel
import scipy
from scipy import stats
from scipy.stats import binom,poisson,norm,t,expon,f
from sklearn.model_selection import train_test_split
import statsmodels
from statsmodels import stats
from statsmodels.stats import weightstats as ssw

In [2]:
df=pd.read_excel('/content/drive/MyDrive/My pitch/CDAC_DataBook.xlsx',sheet_name='faithful')
df.head()

,eruptions,waiting
0,3.600,79
1,1.800,54
2,3.333,74
3,2.283,62
4,4.533,85


In [6]:
erupt=df.eruptions
xbar=np.mean(erupt)
n=len(erupt)
sd=np.std(erupt,ddof=1)
print(xbar)

3.4877830882352936


In [31]:
z=(xbar-3.37)/(sd/n**0.5)

# 1-sample z test/ 1 sample t test:

- Used to check whether the difference betweent the sample mean and claim value can be ignored or not.
- Used to check whether the difference betweent the sample mean and claim value is statistically significant or not

1  h0 u>=3.6
  ha u<3.6

2 cl=95%
los=5%

3 1 sample z test

4 ts p-value

In [14]:
ssw.ztest(erupt,value=3.6,alternative='smaller')


(np.float64(-1.621495827018525), np.float64(0.052455672107759366))

In [17]:
ssw.ztest(erupt,value=3.7,alternative='smaller') # first one z second p-value if p smaller than los(0.05)
norm.ppf(0.05)


np.float64(-1.6448536269514729)

rejection will happen when the magnitude

In [20]:
z=(3.487-3.53)/(sd/n**0.5)
z

-0.6213352289358334

In [24]:
norm.cdf(z)

np.float64(0.2671895395328251)

los=5%
value on x through ppf
if it is less  

start point of 5% on standard normal curve is -1.6448536269514729

and z is -3.066461475706521 and p-value(0.0010830435959524984)

because the z-value is lesser than start point of 5% so we reject the NULL HYPOTHESIS.

Upto -1.645 is okay to be DO NOT REJECT.

---

There is 2 options to choose between Rejection or not to NULL HYPOTHESIS
1. Compare with z-value:

**Rejection will happen if z-value is more than 5%**

z-value is value on x-axis, if it's magnitude is more than Zc then we REJECT

2. Compare with p-value:

p-value is percentage, if it's magnitude is less than LOS then REJECT

In [26]:
ssw.ztest(erupt,value=3.53,alternative='smaller') #NULL HYPOTHESIS WILL NOT BE REJECTED BCOZ P VALUE GREATER THAN 0.05 (LOS)

(np.float64(-0.6100198729369251), np.float64(0.27092432161445423))

In [30]:
#U(mean) will not exceed 3.37
ssw.ztest(erupt,value=3.37,alternative='larger') #NULL hypothesis will be rejected p value is samller 0.05 and and 1.7 is x axis figure

(np.float64(1.7019251649638607), np.float64(0.044384698843295746))

In [28]:
norm.ppf(0.95)

np.float64(1.6448536269514722)